### PR TITLE
{cae}[intel/2018a] OpenFOAM v2.2.2

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2591.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/7e9ebda3fbef856fa2ab4f5cd9e65cfc for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3159.skitty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/14be506b66ef9894f93346e56ba289de for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2081.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/20069ee5452cebcc58b4d35d98a156e9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3157.skitty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/e146f189d93d31cbb3f6e15fc5ba7c0d for a full test report.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in this PR)
cermis - Linux debian 9.4, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/b7a70f531a9308bd8d7e75fb14aa77fb for a full test report.

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.2.2-intel-2018a.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ involving chemical reactions, turbulence and heat transfer, 
+ to solid dynamics and electromagnetics."""
+
+toolchain = {'name': 'intel', 'version': '2018a'}
+
+source_urls = ['http://downloads.sourceforge.net/foam/%(version)s']
+sources = [
+    SOURCE_TGZ,
+    'ThirdParty-%(version)s.tgz',
+]
```
Comment (vanzod) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.2.2-intel-2018a.eb]: Missing checksum

**Comment by vanzod** [comment]
@boegel Ping

**Comment by boegel** [comment]
@vanzod Can you upload a full (debug) log somewhere to dive in?

**Comment by vanzod** [comment]
It looks like it is having problems building `ensightPart.o` and `ensightPartIO.o` because of undefined `isnan`. Linking `libsampling.so` also fails because it cannot find `libconversion`, which causes other linker failures down the building tree.
Attached is the full build log.

[easybuild-run_cmd-fYCT31.log](https://github.com/easybuilders/easybuild-easyconfigs/files/2389189/easybuild-run_cmd-fYCT31.log)

**Comment by boegel** [comment]
@vanzod Looks like we'll need a patch to change `isnan` to `std::isnan`, cfr. https://github.com/cocos2d/cocos2d-x/issues/15111 . Whether or not this fails probably depends on the exact `glibc` version you have.

Are you up to try and patch that? If the other problem still remains afterwards, we can try and figure out that one too...

**Comment by boegel** [comment]
@vanzod The patch we need is probably similar/identical to the one @akesandgren added in #7439?

**Comment by akesandgren** [comment]
The same patch should work, but might need manual patching to get a clean variant.
Don't think there are any more bad isnan's in 2.2.2 than in 3.0.1

**Comment by vanzod** [comment]
@boegel I confirm that adding `OpenFOAM-3.0.1_fix_isnan.patch` as is solves the build issues.

**Comment by boegel** [comment]
@vanzod patch added, please retest?

**Comment by boegel** [comment]
@vanzod re-ping?

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-an04.hpc2n.umu.se - Linux ubuntu 16.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/624263a008bcc3995afe32267d078541 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2610.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/7490648f164b15e92c23ed573bd91ab0 for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**